packages/core/src/main/shims/node/assetCryptography.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {EncryptedAsset} from '../../crypto/root';
+import * as crypto from 'crypto';
+
+const equalHashes = (bufferA: Buffer, bufferB: Buffer): boolean => {
+  const arrayA = new Uint32Array(bufferA);
+  const arrayB = new Uint32Array(bufferB);
+  return arrayA.length === arrayB.length && arrayA.every((value, index) => value === arrayB[index]);
+};
+
+export const decryptAsset = async ({
+  cipherText,
+  keyBytes,
+  sha256: referenceSHA256,
+}: EncryptedAsset): Promise<Buffer> => {
+  const computedSHA256 = crypto
+    .createHash('SHA256')
+    .update(cipherText)
+    .digest();
+
+  if (!equalHashes(computedSHA256, referenceSHA256)) {
```

Comment:
We should always consider the positive case first: 

```ts
if (equalHashes(computedSHA256, referenceSHA256)) { 
  ...
} else { 
 throw new Error('Encrypted asset does not match its SHA-256 hash'); 
}
```

Reply:
:-1: :-1: :-1: